install/installer/aws.py

```diff
@@ -0,0 +1,272 @@
+        keys: (list) AWS_ACCESS_KEY_ID's
+    Returns:
+        Dict[str, Union[str, bool]]:
+        - `{'key': 'AWS_ACCESS_KEY_ID', 'secret': 'AWS_SECRET_ACCESS_KEY', 'created': bool}`
+        - `{}` - if keys not affiliate to user.
+    """
+    for key in keys:
+        if key == login:
+            logger.info(f'Specified creds belong to user "{user}", use it')
+
+            return {
+                'key': login,
+                'secret': password,
+                'created': False,
+            }
+
+    logger.debug(f'Specified keys not belong to user "{user}"')
+    return {}
+
+
+@typechecked
+def create_access_key(
+    iam: boto3.client,
+    user: str,
+) -> Dict[str, Union[str, bool]]:
+    """Create Access Key for exist user.
+
+    Required "iam:CreateAccessKey".
+
+    Args:
+        iam: (boto3.client.IAM) A low-level client representing AWS IAM.
+        user: (str) Cluster.dev user name.
+
+    Returns:
+        Dict[str, Union[str, bool]]:
+        `{'key': 'AWS_ACCESS_KEY_ID', 'secret': 'AWS_SECRET_ACCESS_KEY', 'created': bool}`
+    """
+    response = iam.create_access_key(UserName=user)
+
+    logger.debug('Access keys created')
+    return {
+        'key': response['AccessKey']['AccessKeyId'],
+        'secret': response['AccessKey']['SecretAccessKey'],
+        'created': True,
+    }
+
+
+@typechecked
+def create_user_and_permissions(
+    user: str,
+    login: str,
+    password: str,
+    session: str,
+    release: str,
+) -> Dict[str, Union[str, bool]]:
+    """Create cloud user and attach needed permissions.
+
+    Args:
+        user: (str) Cluster.dev user name.
+        login: (str) Cloud programatic login.
+        password: (str) Cloud programatic password.
+        session: (str) Cloud session token.
+        release: (str) cluster.dev release tag.
+
+    Returns:
+        Dict[str, Union[str, bool]]:
+        `{'key': 'AWS_ACCESS_KEY_ID', 'secret': 'AWS_SECRET_ACCESS_KEY', 'created': bool}`.
+    """
+    iam = boto3.client(
+        'iam',
+        aws_access_key_id=login,
+        aws_secret_access_key=password,
+        aws_session_token=session,
+    )
+
+    # cli_rights_needed_msg
+    msg = (
+        '\n\nRights what you need described here: \n'
+        + f'https://github.com/shalb/cluster.dev/blob/{release}'
+        + '/install/installer_aws_install_req_permissions.json'
+    )
+
+    if user_exist(iam, user, msg):
+        keys = list_user_access_keys(iam, user, msg)
+
+        if not keys:
+            return create_access_key(iam, user)
+
+        creds = check_affiliation_user_and_creds(user, login, password, keys)
+        if creds:
+            return creds
+
+    keys_created = False
+    link = (
+        f'https://github.com/shalb/cluster.dev/blob/{release}'
+        + '/install/installer_aws_install_req_permissions.json'
+    )
+
+    # Create/get policy_arn for user
+    with open('aws_policy.json', 'r') as policy_file:
+        policy = json.dumps(json.load(policy_file))
```

Comment:
What is the point of converting strings JSON back and forth? Validation?
Also, it's nicer to use a higher-level API of pathlib:
```suggestion
    from pathlib import Path
    aws_policy_path = Path('aws_policy.json')
    policy = aws_policy_path.read_text(policy_file)
    # Verify that it's a valid JSON string:
    json.loads(policy)
```

Reply:
Validation is good point, but when I it wrote, main reason was "I need json-string"

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/iam.html#IAM.Client.create_policy

Reply:
But reading the code the file already contains that JSON string so you just need to read the text and that's it.